crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -300,10 +283,97 @@ impl Persistence {
             id: auction_id,
             orders,
             prices,
-            deadline,
             surplus_capturing_jit_order_owners,
         })
     }
+
+    /// Returns the proposed solver solution that won the competition for a
+    /// given auction.
+    ///
+    /// It is expected for a solution to exist, so missing data is considered an
+    /// error.
+    ///
+    /// Returns error for old non-colocated auctions.
+    pub async fn get_winning_solution(
+        &self,
+        auction_id: domain::auction::Id,
+    ) -> Result<domain::competition::Solution, error::Solution> {
+        let _timer = Metrics::get()
+            .database_queries
+            .with_label_values(&["get_competition_winner"])
+            .start_timer();
+
+        let mut ex = self
+            .postgres
+            .pool
+            .begin()
+            .await
+            .map_err(error::Solution::BadCommunication)?;
+
+        let competition = database::settlement_scores::fetch(&mut ex, auction_id)
+            .await
+            .map_err(error::Solution::BadCommunication)?
+            .ok_or(error::Solution::NotFound)?;
+
+        let winner = H160(competition.winner.0).into();
+        let score = competition::Score::new(
+            big_decimal_to_u256(&competition.winning_score)
+                .ok_or(error::Solution::InvalidScore(anyhow::anyhow!(
+                    "database score"
+                )))?
+                .into(),
+        )
+        .map_err(|_| error::Solution::InvalidScore(anyhow::anyhow!("zero score")))?;
```

Comment:
maybe we can have the error also alongisde `zero score`. All errors implement `ToString` trait.

Reply:
added